File: qf-auto-repair-pojo/src/main/java/com/qianfeng/entity/Part.java
package com.qianfeng.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@TableName("parts")
@Data
public class Part {
    @TableId(type = IdType.AUTO)
    private Integer partId;

    @TableField("part_name")
    private String partName;

    @TableField("part_number")
    private String partNumber;

    @TableField("description")
    private String description;

    @TableField("stock_quantity")
    private Integer stockQuantity;

    @TableField("supplier_id")
    private Integer supplierId;

    @TableField("purchase_price")
    private BigDecimal purchasePrice;

    @TableField("sale_price")
    private BigDecimal salePrice;
    // 创建时间
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    // 更新时间
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: src/types/part.ts
export interface PartDTO {
  partId?: number; // 可选字段，用于更新时指定 ID
  partName: string;
  partNumber: string;
  description: string;
  stockQuantity: number;
  supplierId: number;
  purchasePrice: number; // 使用 number 类型表示价格
  salePrice: number; // 使用 number 类型表示价格
}

export interface PartVO {
  partId: number;
  partName: string;
  partNumber: string;
  description: string;
  stockQuantity: number;
  supplierId: number;
  purchasePrice: number; // 使用 number 类型表示价格
  salePrice: number; // 使用 number 类型表示价格
  supplierName: string; // 额外字段：供应商名称
}

export interface PartQueryDTO {
  partId?: number; // 可选字段，用于更新时指定 ID
  partName: string;
  partNumber: string;
  description: string;
  stockQuantity: number;
  supplierId: number;
  purchasePrice: number; // 使用 number 类型表示价格
  salePrice: number; // 使用 number 类型表示价格
  current: number;
  size: number;
}